packages/@aws-cdk/aws-appmesh/lib/virtual-router-listener.ts

```diff
@@ -0,0 +1,122 @@
+import * as cdk from '@aws-cdk/core';
+import { CfnVirtualRouter } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+
+/**
+ * Represents the properties needed to define Listeners for a VirtualRouter
+ */
+export interface RouterListenerProps {
+  /**
+   * Port to listen for connections on
+   *
+   * @default - 8080
+   */
+  readonly port?: number
+}
+
+/**
+ * Properties for a VirtualRouter listener
+ */
+export interface VirtualRouterListenerConfig {
+  /**
+   * Single listener config for a VirtualRouter
+   */
+  readonly listener: CfnVirtualRouter.VirtualRouterListenerProperty;
+}
+
+/**
+ * Represents the properties needed to define listeners for a VirtualRouter
+ */
+export abstract class VirtualRouterListener {
+  /**
+   * Returns an HTTP Listener for a VirtualRouter
+   */
+  public static httpVirtualRouterListener(props: RouterListenerProps = {}): VirtualRouterListener {
```

Comment:
I'm starting to have some second thoughts about the names of these methods 😕. I feel like we're saying `VirtualRouterListener` twice each time (once in the class name, once in the method name).

I wonder if we should keep the class as `VirtualRouterListener`, but change the methods to something like `http()` / `http2()` / `grpc()`.

Thoughts on this?

Reply:
Seems reasonable. I'd opt for less verbosity here. Will need to update for VirtualGateways as well

Reply:
Yep, that's fine (we'll probably have to change the `*Props` there to `*Options` anyway because of my above comment).